exasol_sphinx_github_pages_generator/deployer.py

```diff
@@ -0,0 +1,192 @@
+import subprocess
+import sys
+from pathlib import Path
+from subprocess import run
+import shutil
+import os
+
+
+# TODO remove debug outputs
+# todo add typing, docu comments
+
+class GithubPagesDeployer:
+    def __init__(self, source_dir, module_path, target_branch, push_origin, push_enabled, source_branch,
+                 current_commit_id, tempdir):
+        self.source_dir = source_dir
+        self.module_path = module_path
+        self.target_branch = target_branch
+        self.push_origin = push_origin
+        self.push_enabled = push_enabled
+        self.source_branch = source_branch
+        self.current_commit_id = current_commit_id.stdout
+        self.worktree_paths = {"target_worktree": tempdir + "/worktrees/worktree_target",
+                              "source_worktree": tempdir + "/worktrees/worktree_source"}
+        self.build_dir = tempdir + "/build"
+        self.intermediate_dir = tempdir + "/intermediate"
+        self.staged_changes = False
+
+    def check_out_source_branch_as_worktree(self,
+                                            source_branch_commit_id):
+        source_branch_exists_remote = run(["git", "show-branch", f"remotes/origin/{self.source_branch}"],
+                                          capture_output=True, text=True)
+        print("source_branch_exists_remote : " + str(source_branch_exists_remote))
+        if source_branch_exists_remote.returncode == 0:
+            try:
+                # "If <branch> does exist, it will be checked out in the new working tree, if it’s not checked out
+                # anywhere else, otherwise the command will refuse to create the working tree (unless --force is used)."
+                run(["git", "worktree", "add", self.worktree_paths["source_worktree"],
+                     self.source_branch, "--force"], check=True)
+                print(f"Successfully added new temp worktree for source branch {self.source_branch}, "
+                      f"and checked out (Local or stashed changes will be ignored in this build).")
+                os.chdir(f"{self.worktree_paths['source_worktree']}/doc")
+                current_branch = run(["git", "rev-parse", "--abbrev-ref", "HEAD"], capture_output=True, text=True,
+                                     check=True)
+                print("new current_branch/detected source branch : " + str(current_branch.stdout))
+            except subprocess.CalledProcessError:
+                sys.exit("Problem with adding worktree for source")
```

Comment:
printing error information to stderr `print("error msg", file=sys.stderr)` and exiting with a exit value `!=0` afterwards `sys.exit(1)` afterwards is  more clear in general (especially to non python coders). That's why I personally would
take the extra line in favor of maintainability and complexity for other readers.

Reply:
Same applies for the other exit statements with `returncode != 0`

Reply:
sys.exit can take an arbitrary object, if this is the case it prints the object  to stderr and returns the return_code 1

https://docs.python.org/3/library/sys.html#sys.exit

Reply:
However, I had to check the documentation as well for that.

Reply:
It functionally does the same thing. i think its just a matter of preference. i think the sys.exit(message) looks cleaner.